sinks/influxdb/driver.go

```diff
@@ -37,72 +37,130 @@ type influxdbSink struct {
 	}
-	// Add timestamp.
-	columns = append(columns, "time")
-	values = append(values, timeseries.Point.End.Unix())
-	// Ass sequence number
-	columns = append(columns, "sequence_number")
-	values = append(values, self.seqNum.Get(seriesName))
-
-	return self.newSeries(seriesName, columns, values)
+	return nil
 }
 
-func (self *influxdbSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+func (sink *influxdbSink) storeTimeseriesWithPrecision(timeseriesLists []*[]sink_api.Timeseries, precision sink_api.TimePrecision) error {
 	dataPoints := []*influxdb.Series{}
-	for index := range timeseries {
-		dataPoints = append(dataPoints, self.metricToSeries(&timeseries[index]))
+	for _, timeseriesList := range timeseriesLists {
+		if timeseriesList != nil {
+			for _, timeseries := range *timeseriesList {
+				if timeseries.TimePrecision == precision {
+					if !sink.avoidColumns {
+						dataPoints = append(dataPoints, sink.timeSeriesToInfluxdbSeries(&timeseries))
+					} else {
+						for _, point := range timeseries.Points {
+							dataPoints = append(dataPoints, sink.pointToInfluxdbSeriesNoColumn(timeseries.SeriesName, point))
+						}
+					}
+				}
+			}
+		}
 	}
+
 	// TODO: Group all datapoints belonging to a metric into a single series.
 	// TODO: Record the average time taken to flush data.
-	err := self.client.WriteSeriesWithTimePrecision(dataPoints, influxdb.Second)
-	if err != nil {
-		glog.Errorf("failed to write stats to influxDB - %s", err)
-		self.recordWriteFailure()
+	if len(dataPoints) > 0 {
+		err := sink.client.WriteSeriesWithTimePrecision(dataPoints, getInfluxdbPrecision(precision))
+		if err != nil {
+			glog.Errorf("failed to write time series to influxDB - %s", err)
+			for _, dataPoint := range dataPoints {
+				glog.Errorf("  detail: failed to write time series (%s) to influxDB with columns (%s)", dataPoint.Name, dataPoint.Columns)
+			}
+			sink.recordWriteFailure()
+			return err
+		}
+		glog.V(1).Info("Succesfully wrote %q timeseries to influxDB", precision)
 	}
-	glog.V(1).Info("flushed stats to influxDB")
-	return err
+	return nil
 }
 
-// Returns a new influxdb series.
-func (self *influxdbSink) newSeries(seriesName string, columns []string, points []interface{}) *influxdb.Series {
-	out := &influxdb.Series{
+func getInfluxdbPrecision(precision sink_api.TimePrecision) influxdb.TimePrecision {
+	switch precision {
+	case sink_api.Second:
+		return influxdb.Second
+	case sink_api.Millisecond:
+		return influxdb.Millisecond
+	case sink_api.Microsecond:
+		return influxdb.Microsecond
+	}
+	return influxdb.Second
+}
+
+func (sink *influxdbSink) timeSeriesToInfluxdbSeries(timeseries *sink_api.Timeseries) *influxdb.Series {
+	seriesName := timeseries.SeriesName
+
+	columns := []string{}
+	columns = append(columns, "time")            // Column 0 header
+	columns = append(columns, "value")           // Column 1 header
+	columns = append(columns, "sequence_number") // Column 2 header
```

Comment:
How about sharing setting up of these standard columns between the two methods here?